avm/ptn/dev-ops/cicd-agents-and-runners/main.bicep

```
@@ -0,0 +1,1030 @@
+
+// ================ //
+// Parameters       //
+// ================ //
+
+@description('Optional. Location for all resources.')
+param location string = resourceGroup().location
+
+@description('Required. Naming prefix to be used with naming the deployed resources.')
+param namingPrefix string
+
+@description('Required. The compute target for the private runners.')
+param computeTypes computeTypesType
+
+@description('Required. The self-hosted runner configuration. This can be either GitHub or Azure DevOps.')
+param selfHostedConfig selfHostedRunnerType
+
+@description('Required. The networking configuration.')
+param networkingConfiguration networkType
+
+@description('Optional. Whether to use private or public networking for the Azure Container Registry.')
+param privateNetworking bool = true
+
+@description('Optional. Enable/Disable usage telemetry for module.')
+param enableTelemetry bool = true
+
+// ================ //
+// Variables        //
+// ================ //
+
+var imagePaths = [
+  {
+    platform: 'azuredevops-container-app'
+    imagePath: 'azure-devops-agent-aca'
+  }
+  {
+    platform: 'github-container-app'
+    imagePath: 'github-runner-aca'
+  }
+  {
+    platform: 'azuredevops-container-instance'
+    imagePath: 'azure-devops-agent-aci'
+  }
+  {
+    platform: 'github-container-instance'
+    imagePath: 'github-runner-aci'
+  }
+]
+
+// To be updated
+var githubURL = 'https://github.com/sebassem/bicep-registry-modules#avm-ptn-lz-private-runners:avm/ptn/dev-ops/cicd-agents-and-runners/container-images'
+
+var acaGitHubRules = (selfHostedConfig.selfHostedType == 'github')
+  ? [
+      {
+        name: 'github-runner-scaling-rule'
+        type: 'github-runner'
+        metadata: {
+          owner: selfHostedConfig.githubOrganization
+          repos: selfHostedConfig.githubRepository
+          targetWorkflowQueueLength: selfHostedConfig.?targetWorkflowQueueLength ?? '1'
+          runnerScope: selfHostedConfig.?runnerScope ?? 'repo'
+        }
+        auth: [
+          {
+            secretRef: 'personal-access-token'
+            triggerParameter: 'personalAccessToken'
+          }
+        ]
+      }
+    ]
+  : []
+
+var acaGitHubSecrets = (selfHostedConfig.selfHostedType == 'github')
+  ? [
+      {
+        name: 'personal-access-token'
+        value: any(selfHostedConfig.personalAccessToken)
+      }
+    ]
+  : []
+
+var acaGitHubEnvVariables = (selfHostedConfig.selfHostedType == 'github')
+  ? [
+      {
+        name: 'RUNNER_NAME_PREFIX'
+        value: selfHostedConfig.?runnerNamePrefix ?? 'gh-runner'
+      }
+      {
+        name: 'REPO_URL'
+        value: gitHubRunnerURL
+      }
+      {
+        name: 'RUNNER_SCOPE'
+        value: selfHostedConfig.?runnerScope ?? 'repo'
+      }
+      {
+        name: 'EPHEMERAL'
+        value: selfHostedConfig.?ephemeral ?? 'true'
+      }
+      {
+        name: 'ORG_NAME'
+        value: selfHostedConfig.gitHubOrganization
```

```suggestion
        value: selfHostedConfig.?gitHubOrganization
```

Are you sure you need the variable
```
var gitHubOrganization = selfHostedConfig.selfHostedType == 'github' ? selfHostedConfig.githubOrganization : null
```
as the above is evaluating into the same, that is `null`. The same is true for the `gitHubRepository` variable.

---

Resolved